marklogic-client-api/src/test/java/com/marklogic/client/test/datamovement/ProgressListenerTest.java

```diff
@@ -0,0 +1,175 @@
+package com.marklogic.client.test.datamovement;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.*;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class ProgressListenerTest extends Assert {
+
+	private ProgressListener listener;
+	private TestConsumer consumer = new TestConsumer();
+
+	@Test
+	public void batchesInOrder() throws Exception {
+		listener = new ProgressListener(4, consumer);
+		listener.initializeListener(null);
+		// Ensure that the total amount of time will be greater than zero for when it's tested below
+		Thread.sleep(100);
+
+		listener.processEvent(new FakeQueryBatch(2));
+		assertFalse(consumer.lastProgressUpdate.isComplete());
+
+		listener.processEvent(new FakeQueryBatch(4));
+		assertTrue(consumer.lastProgressUpdate.isComplete());
+
```

Comment:
another test of FakeQueryBatch(0) might expose the problem with above isComplete()